cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c

```diff
@@ -108,48 +108,23 @@ zephyr_img_mgmt_flash_area_id(int slot)
         break;
 
     default:
-        assert(0);
-        fa_id = FLASH_AREA_ID(image_1);
+        fa_id = -1;
         break;
     }
 
     return fa_id;
 }
 
+/**
+ * The function will check if second(ary) slot is available, and allowed,
+ * for DFU; if checks area positive, then area ID is returned, -1 is returned
+ * otherwise.
+ */
 static int
-img_mgmt_find_best_area_id(void)
+img_mgmt_get_second_slot_area_id(void)
 {
-    struct image_version ver;
-    int best = -1;
-    int i;
-    int rc;
-
-    for (i = 0; i < 2; i++) {
-        rc = img_mgmt_read_info(i, &ver, NULL, NULL);
-        if (rc < 0) {
-            continue;
-        }
-        if (rc == 0) {
-            /* Image in slot is ok. */
-            if (img_mgmt_slot_in_use(i)) {
-                /* Slot is in use; can't use this. */
-                continue;
-            } else {
-                /*
-                 * Not active slot, but image is ok. Use it if there are
-                 * no better candidates.
-                 */
-                best = i;
-            }
-            continue;
-        }
-        best = i;
-        break;
-    }
-    if (best >= 0) {
-        best = zephyr_img_mgmt_flash_area_id(best);
-    }
-    return best;
+    /* In case of Zephyr slot 0 is always unavailable */
+    return img_mgmt_slot_in_use(1) == 0 ? zephyr_img_mgmt_flash_area_id(1) : -1;
```

Comment:
I'm afraid that this will made update impossible for DIRECT-XIP MCUboot compatible mode, when either primary or secondary slot might be the update target.